nntrainer/layers/multi_head_attention_layer.cpp

```diff
@@ -77,10 +77,415 @@ enum AttentionParams {
   transposed_attention_output,
 };
 
-void MultiHeadAttentionLayer::finalizeCommon(InitLayerContext &context) {}
+void MultiHeadAttentionLayer::finalizeCommon(InitLayerContext &context) {
+  if (context.getNumInputs() < 2 || context.getNumInputs() > 4) {
+    throw std::runtime_error("Multi head Attention layer needs 2-4 inputs.");
+  }
+
+  NNTR_THROW_IF(
+    std::get<props::ReferenceFramework>(multi_head_attention_props).empty(),
+    std::invalid_argument)
+    << "reference_framework property is not provided for layer "
+    << context.getName();
+  const Framework ref_framework =
+    std::get<props::ReferenceFramework>(multi_head_attention_props).get();
+
+  NNTR_THROW_IF(std::get<props::NumHeads>(multi_head_attention_props).empty(),
+                std::invalid_argument)
+    << "num_heads property is not provided for layer " << context.getName();
+  const unsigned int num_heads =
+    std::get<props::NumHeads>(multi_head_attention_props).get();
+
+  NNTR_THROW_IF(std::get<props::QueryDim>(multi_head_attention_props).empty(),
+                std::invalid_argument)
+    << "query_dim property is not provided for layer " << context.getName();
+  const unsigned int query_dim_prop =
+    std::get<props::QueryDim>(multi_head_attention_props).get();
+
+  if (std::get<props::KeyDim>(multi_head_attention_props).empty()) {
+    std::get<props::KeyDim>(multi_head_attention_props).set(query_dim_prop);
+  } else {
+    NNTR_THROW_IF(ref_framework == Framework::tensorflow, std::invalid_argument)
```

Comment:
I think it might be better take one way to get the property as I describe https://github.com/nnstreamer/nntrainer/pull/1956/files#r926149987

Reply:
As I commented in https://github.com/nnstreamer/nntrainer/pull/1956/files#r934121659, ProjectedkeyDim is not alias of KeyDim but as you recommended if the ProjectedKeyDim is not given then set the default value using the Keydim.